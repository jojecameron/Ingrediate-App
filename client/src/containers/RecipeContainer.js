import React, { Component } from 'react';


class RecipeContainer extends Component {
    constructor(props) {
        super(props);
    }

    // Handle recipe components displaying in container

    render() {
        return( 
            <div className="container"> 
        
            </div> 
        )
    }
}


export default RecipeContainer;