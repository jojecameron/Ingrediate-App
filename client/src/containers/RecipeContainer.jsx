import React, { useState } from 'react';
import Recipe from '../components/Recipe';

const RecipeContainer = (props) => {

  // Handle recipe components displaying in container
  
    const recipes = props.recipeList.map((recipe, i) => {
      //   console.log('recipe being generated',recipe.recipe, i);
      const splitRecipe = recipe.recipe.split('|');
      //   console.log('this is splitRecipe[1]',splitRecipe[1]);
      return (
        <Recipe
          key={`Recipe${i}`}
          recipeIndex={i}
          deleteRecipe={props.deleteRecipe}
          favoriteRecipe={props.favoriteRecipe}
          recipeTitle={splitRecipe[0]}
          recipeText={splitRecipe[1]}
          recipeLinkTitle={splitRecipe[2]}
          recipeLink={splitRecipe[3]}
        />
      );
    });

    return (
      <div className="RecipeContainer">
        <div className="containerTitle">
          <h1>
            <em>Recipes</em>
          </h1>
          <hr />
        </div>
        <div className="recipeDisplay">{recipes}</div>
      </div>
    );
}

export default RecipeContainer;
