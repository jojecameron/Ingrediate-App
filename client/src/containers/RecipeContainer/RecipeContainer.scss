.RecipeContainer {
  background-color: $offWhite;
  border-radius: 15px;
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.2);
  height: fit-content;
  margin-top: 50px;
  min-height: 80vh;
  width: 87.5%;

  .recipeDisplay {
    align-items: center;
    display: flex;
    flex-flow: wrap;
    justify-content: center;
  }
  h1 {
    color: $primaryGreen;
    font-size: 30px;
    padding-left: 20px;
  }
}
