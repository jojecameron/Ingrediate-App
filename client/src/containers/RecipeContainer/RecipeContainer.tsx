import { useState } from 'react';
import { Recipe } from '../../components';
import { RecipeContainerProps, Favorite } from '../../types';

const RecipeContainer = (props: RecipeContainerProps): JSX.Element => {
  const [favoriteMode, setFavoriteMode] = useState<boolean>(false);

  const {
    recipeList,
    deleteRecipe,
    favoriteRecipe,
    favoriteRecipes,
    setFavoriteRecipes,
    updateRecipeTitle
  } = props;

  const recipes = recipeList.map((recipe, i) => {
    return (
      <Recipe
        key={recipe.id}
        id={recipe.id}
        deleteRecipe={deleteRecipe}
        favoriteRecipe={favoriteRecipe}
        recipeTitle={recipe.recipeTitle}
        recipeText={recipe.recipeText}
        recipeLinkTitle={recipe.recipeLinkTitle}
        recipeLink={recipe.recipeLink}
        updateRecipeTitle={updateRecipeTitle}
      />
    );
  });

  const favorites = favoriteRecipes.map((favorite) => {
    return (
      <Recipe
        key={favorite.id}
        id={favorite.id}
        deleteRecipe={deleteRecipe}
        favoriteRecipe={favoriteRecipe}
        recipeTitle={favorite.recipeTitle}
        recipeText={favorite.recipeText}
        recipeLinkTitle={favorite.recipeLinkTitle}
        recipeLink={favorite.recipeLink}
        updateRecipeTitle={updateRecipeTitle}
        favorite={true}
      />
    );
  });

  return (
    <>
      <div className="containerTitle">
        {!favoriteMode ? (
          <>
            <h1 onClick={() => setFavoriteMode(false)} className="active">
              <em>Recipes</em>
            </h1>
            <h1 onClick={() => setFavoriteMode(true)} className="inactive">
              <em>Favorites</em>
              {favoriteRecipes.length > 0 ? (
                <span id="numFavorites">{favoriteRecipes.length}</span>
              ) : null}
            </h1>
          </>
        ) : (
          <>
            <h1 onClick={() => setFavoriteMode(false)} className="inactive">
              <em>Recipes</em>
            </h1>
            <h1 onClick={() => setFavoriteMode(true)} className="active">
              <em>Favorites</em>
            </h1>
          </>
        )}
      </div>
      <div className="RecipeContainer">
        <div className="recipeDisplay">
          {!favoriteMode ? recipes : favorites}
        </div>
      </div>
    </>
  );
};

export default RecipeContainer;
