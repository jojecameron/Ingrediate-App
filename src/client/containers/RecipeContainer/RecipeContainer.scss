.containerTitle {
  display: flex;
  margin-bottom: 0px;
  margin-top: 40px;
  width: 87.5%;

  h1 {
    background-color: $tertiaryGreen;
    border-radius: 15px 15px 0px 0px;
    color: $primaryGreen;
    display: flex;
    font-family: $mainFont;
    font-size: 30px;
    justify-content: space-between;
    margin: 0px;
    padding: 10px 0px 10px 20px;
    padding-left: 20px;
    width: 50%;
  }
  h1:hover {
    background-color: rgba(0, 0, 0, 0.2);
    transition-duration: 100ms;
  }
  .active {
    background-color: $offWhite;
    box-shadow: 2px 8px 15px rgba(0, 0, 0, 0.2);
    transition-duration: 100ms;
  }
  .active:hover {
    background-color: $offWhite;
  }
  .inactive {
    background-color: $primaryGreen;
    border: $offWhite solid 1px;
    color: $offWhite;
    cursor: pointer;
    transition-duration: 100ms;
  }
  .inactive:hover {
    background-color: rgba(0, 0, 0, 0.2);
    border: $offWhite solid 1px;

    #numFavorites {
      color: darkgreen;
    }
  }

  #numFavorites {
    background-color: $offWhite;
    border-radius: 10px;
    color: $primaryGreen;
    font-size: 20px;
    margin-right: 15px;
    padding: 5px;
    text-align: center;
    width: 20px;
  }
}

.RecipeContainer {
  background-color: $offWhite;
  border-radius: 0px 0px 15px 15px;
  box-shadow: 0px 15px 15px rgba(0, 0, 0, 0.2);
  display: flex;
  flex-direction: column;
  height: fit-content;
  margin-top: 0px;
  min-height: 80vh;
  width: 87.5%;

  .recipeDisplay {
    align-items: center;
    background-color: $offWhite;
    display: flex;
    flex-flow: wrap;
    justify-content: center;
  }
  hr {
    margin: 0px;
    width: auto;
  }
}
