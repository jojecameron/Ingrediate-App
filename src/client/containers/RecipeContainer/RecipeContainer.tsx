import { Recipe } from '../../components';
import { RecipeContainerProps } from '../../types';

const RecipeContainer = (props: RecipeContainerProps): JSX.Element => {
  const {
    recipeList,
    deleteRecipe,
    favoriteRecipe,
    favoriteRecipes,
    favoriteMode,
    setFavoriteMode,
    setRecipeModal,
  } = props;

  const recipes = recipeList.map((recipe, i) => {
    return (
      <Recipe
        key={recipe.id}
        id={recipe.id}
        deleteRecipe={deleteRecipe}
        favoriteRecipe={favoriteRecipe}
        recipeTitle={recipe.recipeTitle}
        recipeText={recipe.recipeText}
        setRecipeModal={setRecipeModal}
      />
    );
  });

  const favorites = favoriteRecipes.map((favorite) => {
    return (
      <Recipe
        key={favorite.id}
        id={favorite.id}
        deleteRecipe={deleteRecipe}
        favoriteRecipe={favoriteRecipe}
        recipeTitle={favorite.recipeTitle}
        recipeText={favorite.recipeText}
        favorite={true}
        setRecipeModal={setRecipeModal}
      />
    );
  });

  return (
    <>
      <div className="containerTitle">
        {!favoriteMode ? (
          <>
            <h1 onClick={() => setFavoriteMode(false)} className="active">
              <em>Recipes</em>
            </h1>
            <h1 onClick={() => setFavoriteMode(true)} className="inactive">
              <em>Favorites</em>
              {favoriteRecipes.length > 0 ? (
                <span id="numFavorites">{favoriteRecipes.length}</span>
              ) : null}
            </h1>
          </>
        ) : (
          <>
            <h1 onClick={() => setFavoriteMode(false)} className="inactive">
              <em>Recipes</em>
            </h1>
            <h1 onClick={() => setFavoriteMode(true)} className="active">
              <em>Favorites</em>
            </h1>
          </>
        )}
      </div>
      <div className="RecipeContainer">
        <div className="recipeDisplay">
          {!favoriteMode ? recipes : favorites}
        </div>
      </div>
    </>
  );
};

export default RecipeContainer;
