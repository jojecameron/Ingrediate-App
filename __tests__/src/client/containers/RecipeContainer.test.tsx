import React from 'react';
import { render } from '@testing-library/react';
import RecipeContainer from '../../../../src/client/containers/RecipeContainer/RecipeContainer';

const mockProps = {
  recipeList: [],
  deleteRecipe: jest.fn(),
  favoriteRecipe: jest.fn(),
  favoriteRecipes: [],
  setFavoriteRecipes: jest.fn(),
  updateRecipeTitle: jest.fn(),
  favoriteMode: false,
  setFavoriteMode: jest.fn(),
  setRecipeModal: jest.fn(),
};

test('renders a basic snapshot', () => {
  const { container } = render(<RecipeContainer {...mockProps} />);
  expect(container).toMatchSnapshot();
});
