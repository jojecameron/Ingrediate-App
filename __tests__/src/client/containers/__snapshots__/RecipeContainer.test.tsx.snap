// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic snapshot 1`] = `
<div>
  <div
    class="containerTitle"
  >
    <h1
      class="active"
    >
      <em>
        Recipes
      </em>
    </h1>
    <h1
      class="inactive"
    >
      <em>
        Favorites
      </em>
    </h1>
  </div>
  <div
    class="RecipeContainer"
  >
    <div
      class="recipeDisplay"
    />
  </div>
</div>
`;
